orchestrator/careplancontributor/service.go

```diff
@@ -117,6 +145,112 @@ func (s Service) handleProxyToEPD(writer http.ResponseWriter, request *http.Requ
 	proxy.ServeHTTP(writer, request)
 }
 
+func (s Service) handleProxyToFHIR(writer http.ResponseWriter, request *http.Request) error {
+	// Authorize requester before proxying FHIR request
+	// Data holder must verify that the requester is part of the CareTeam by checking the URA
+	// Validate by retrieving the CarePlan from CPS, use URA in provided token to validate against CareTeam
+	// CarePlan should be provided in X-SCP-Context header
+	xScpContext := request.Header["X-Scp-Context"]
+	if len(xScpContext) == 0 {
+		// Check if header has different capitalisation, it may be changed by Go canonicalization
+		xScpContext = request.Header["X-SCP-Context"]
+		if len(xScpContext) == 0 {
+			err := errors.New("X-SCP-Context header value must be set")
+			http.Error(writer, err.Error(), http.StatusUnauthorized)
+			return err
+		}
+	}
+	if len(xScpContext) != 1 {
+		return errors.New("X-SCP-Context header must only contain one value")
+	}
+	carePlanURL := xScpContext[0]
+	if carePlanURL == "" {
+		err := errors.New("X-SCP-Context header value must be set")
+		http.Error(writer, err.Error(), http.StatusUnauthorized)
+		return err
+	}
+	u, err := url.Parse(carePlanURL)
+	if err != nil {
+		return err
```

Comment:
We need some kind of validation of the URLs. We do have a CarePlanService URL configured for the CarePlanContributor right? Probably check whether the configured CPS URL starts with the SCP context URL from the header? That mitigates pretty much any "injected" CPS URL attacks.

Reply:
Good idea, that sounds reasonable